### PR TITLE
refactor: remove `UnaryOperator` in `ErasedAst.Expr.Unary`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/5531

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala [hunk 1]
```diff
@@ -1189,13 +1188,12 @@ object GenExpression {
...
                                jumpLabels: Map[Symbol.LabelSym, Label],
                                entryPoint: Label,
-                               op: UnaryOperator,
                                sop: SemanticOperator)(implicit root: Root, flix: Flix): Unit = {
     // Adding source line number for debugging
     addSourceLine(visitor, e.loc)
 
     compileExpression(e, visitor, currentClassType, jumpLabels, entryPoint)
-    op match {
+    SemanticOperatorOps.toUnaryOp(sop, e.loc) match {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: Oh no. This is not allowed :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: The point is to make the backend indepedent of Unary/Binary OP

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: Oh I just thought you wanted it removed from the AST node

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: Ok, my bad. I did not explain well.

Think of it this way:

Unary/BinaryOp represents a "syntactic" operator, e.g +.
SemanticOp represents an actual operator, e.g. +_Int32 (addition specifically on Int32s).

We want to refactor the backend to not rely on the syntactic ones.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: So what you have done so far is fine, but we need to patmatch on the sop.

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:1196]: Roger :+1:

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**